airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -140,6 +141,18 @@ public AirbyteCatalog discover(JsonNode config) throws Exception {
     return checkOperations;
   }
 
+  @Override
+  public AutoCloseableIterator<AirbyteMessage> read(JsonNode config, ConfiguredAirbyteCatalog catalog, JsonNode state) throws Exception {
```

Comment:
can we add a test for this?

also 90% sure that once you add this all the tests will fail because the tests don't handle setting the replication slot. it's commented out i think